keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -196,22 +223,43 @@ def __init__(
             )
         )
 
-        self.feature_pyramid = layers_lib.FeaturePyramid()
-        prior_probability = keras.initializers.Constant(
-            -np.log((1 - 0.01) / 0.01)
-        )
+        self.feature_pyramid = feature_pyramid
+        self.classification_head = classification_head
+        self.box_head = box_head
+        self.build(backbone.input_shape)
 
-        self.classification_head = (
-            classification_head
-            or layers_lib.PredictionHead(
-                output_filters=9 * num_classes,
-                bias_initializer=prior_probability,
+    @staticmethod
+    def build_forward(
```

Comment:
This feels a bit hacky to me, but maybe it's unavoidable?
If we do decide we want to go with this approach, I think this at least warrants a comment linking to a bug in Keras core describing the issue we're working around by doing this.

wdyt @LukeWood ?

Reply:
Why is it unavoidable? Sounds like I'm missing something

Reply:
Which part do you find hacky?  The functional subclassing?  The build_forwards method?

Reply:
The part that I find hacky is that we have this separate method for building forward. The part that I meant was unavoidable was defining the forward pass in a method for construction time, but Jonathan points out that we can just do this in the constructor a la Functional model style.

That option SGTM (and bonus points that apparently NLP does that too!)

Reply:
Seems we have consensus for doing it in line in the constructor.  Will update and make sure weights still load.

Reply:
Fixed